.gitlab-ci.yml

```diff
@@ -233,6 +233,17 @@ windows32:
   variables:
     ARCH: "32"
 
+nixos:
+  image: nixorg/nix:latest # Minimal NixOS image which doesn't even contain git
+  stage: test
+  variables:
+    GIT_STRATEGY: none
+  dependencies: [] # We don't need to download build artifacts
+  before_script: [] # We don't want to use the shared 'before_script'
+  script:
+    - nix build -f "$CI_PROJECT_URL/-/archive/$CI_COMMIT_SHA.tar.gz" --arg doCheck false
+    # We can only test the build and not the test-suite until #5975 is fixed.
```

Comment:
We can install then run the test suite like gitlab does.

Reply:
That's not really a supported workflow in Nix. That would be a bit hackish. I prefer to wait and see if Dune is really going to solve this problem. If not, I'll propose to remove the problematic tests.

Reply:
Actually I was wrong: this is supported by Nix (https://nixos.org/nixpkgs/manual/#ssec-installCheck-phase). So I'll try that indeed.

Reply:
> The installCheck phase checks whether the package was installed correctly by running its test suite against the installed directories. The default installCheck calls make installcheck. 

We could probably add such a target to the Makefile. It would simplify the gitlab yml too.

Reply:
Indeed. And this reinforces my conviction that it is wrong that currently the `check` target fails if configured with `-prefix`. Anyways, for now I'm remapping `installCheck` to use the `check` target and I'm getting pretty close to being able to run the test-suite successfully.